### PR TITLE
Feature/create event criterias full done

### PR DESCRIPTION
**Pull request by gor017**

### CODE REVIEW

File: src/axiosData.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import axios from 'axios'
+
+const axiosData = axios.create({
+  baseURL: 'http://161.35.144.15:3030',
```
Comment (Arthur02V) [inline src/axiosData.tsx:4]: move to env

File: src/pages/events/createEvents/CreateEventFirstPage.tsx [hunk 2]
```diff
@@ -60,9 +60,10 @@ const CreateEventFirstPage = () => {
     position: 'Position',
     positionValue: '',
     date: 'Hire date',
-    dateValue: undefined,
+    dateValue: '2022-01-01',
```
Comment (Arthur02V) [inline src/pages/events/createEvents/CreateEventFirstPage.tsx:63]: set to select todays value as default

File: src/pages/events/createEvents/CreateEventFirstPage.tsx [hunk 3]
```diff
@@ -366,17 +391,19 @@ const CreateEventFirstPage = () => {
                     }}
                   />
                   <TextField
+                    inputProps={{className: classes.inputText}}
                     name={'dateValue'}
                     label={item.date}
                     type="date"
-                    defaultValue="2022-04-21"
+                    // defaultValue="2022-04-21"
```
Comment (Arthur02V) [inline src/pages/events/createEvents/CreateEventFirstPage.tsx:398]: remove and don't use hardCoded data

**Review by Arthur02V — CHANGES_REQUESTED** [review]
Minor changes requested

**Comment by ArthurO2002** [comment]
Chishta asum ^

File: src/pages/events/createEvents/CreateEventFirstPage.tsx (hunk 3 above)
Comment (gor017) [reply src/pages/events/createEvents/CreateEventFirstPage.tsx]: done

File: src/axiosData.tsx (hunk 1 above)
Comment (gor017) [reply src/axiosData.tsx]: done

File: .env [hunk 4]
```diff
@@ -5,3 +5,4 @@ REACT_APP_STORAGE_BUCKET=evaluation-513c5.appspot.com
 REACT_APP_MESSAGING_SENDER_ID=948069959400
 REACT_APP_APP_ID=1:948069959400:web:d8884376352ea97e17eb5d
 REACT_APP_USER_SERVICE_ENDPOINT=http://161.35.144.15:3000/
+REACT_APP_DATA_ENDPOINT=http://161.35.144.15:3030/
```
Comment (ArthurO2002) [inline .env:8]: Why is env pushed?????

add to gitignore

File: src/pages/events/createEvents/CriteriasPaper.tsx [hunk 5]
```diff
@@ -23,19 +32,266 @@ import CheckedIcon from '../../../assets/images/Icons/CheckedIcon'
...
+    },
+    {
+      values: {
+        firstValue: 1,
+        lastValue: 20,
+      },
+      text: 'From 1 point - to 20 points',
+      id: uuidv4(),
+    },
+  ])
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/CriteriasPaper.tsx:96]: Why adding data from FE, get it from BE

File: src/pages/events/createEvents/CriteriasPaper.tsx [hunk 6]
```diff
@@ -48,190 +304,627 @@ const CriteriasPapers = () => {
...
+                          }
+                        })
+                        setNewSubFilters(newArr)
+                      }}
+                      InputProps={{
+                        classes: {notchedOutline: classes.noBorder},
+                      }}
+                      value={newSubFilter.name}
+                      className={classes.outlinedBasic}
+                      placeholder="Subfilter name"
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/CriteriasPaper.tsx:893]: Component is too big, can we move parts to other components?

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]
LGTM,some things to change

File: src/pages/events/createEvents/CriteriasPaper.tsx (hunk 6 above)
Comment (gor017) [reply src/pages/events/createEvents/CriteriasPaper.tsx]: done

File: src/pages/events/createEvents/createEventCriteria/BonusPercentage.tsx [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+  const changeEditable = (id: string) => {
+    const newArray = params.evaluateesList.map((item: IEvaluatee) => {
+      if (item.id === id) {
+        return {...item, isEditable: true}
+      }
+      return item
+    })
+    params.setEvaluateesList(newArray)
+  }
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/createEventCriteria/BonusPercentage.tsx:24]: you can use  onChangeEvaluateeData to not write double function

File: src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+    event: React.MouseEvent<HTMLButtonElement>,
+    id: React.MouseEvent<HTMLButtonElement> | string,
+  ) => {
+    setAnchorElCriteria(event.currentTarget)
+    const getSubcriteria = async () => {
+      const foundCriteria = params.criteriaData.find((el: ICriteriaPaperInfo) => el.id === id)
+      let data: ISubCriteria[] = []
+      if (foundCriteria !== undefined && foundCriteria.subCriteria.length) {
+        data = foundCriteria.subCriteria
+        await setOpenedSubCriterias(data)
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx:75]: no await

File: src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+        result: true,
+      },
+    ])
+  }
+
+
+  return (
+<Box>
+    <Grid columnSpacing={2} rowSpacing={2} container>
+      {params.criteriaData?.map((item: ICriteriaPaperInfo, index: number) => (
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx:110]: Fix tabulation

File: src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+                <Checkbox
+                  checked={item.criteria}
+                  onChange={(e) => {
+                    const newObj = {...item, criteria: e.target.checked}
+                    const newArr = params.criteriaData.map((newItem: ICriteriaPaperInfo) => {
+                      if (newItem.id === newObj.id) {
+                        return newObj
+                      } else {
+                        return newItem
+                      }
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx:124]: return without else

File: src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx [hunk 11]
```diff
@@ -0,0 +1,312 @@
...
+                    checked={el.result}
+                    onChange={(e) => {
+                      const newObj = {...el, result: e.target.checked}
+                      const newArrSubcriteria = item.subCriteria.map(
+                        (newItem: ISubCriteria) => {
+                          if (newItem.id === newObj.id) {
+                            return newObj
+                          } else {
+                            return newItem
+                          }
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx:159]: same

File: src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx [hunk 12]
```diff
@@ -0,0 +1,312 @@
...
+                        },
+                      )
+                      const bigObj = {...item, subCriteria: newArrSubcriteria}
+                      const newBigArr = params.criteriaData.map((a: ICriteriaPaperInfo) => {
+                        if (bigObj.id === a.id) {
+                          return bigObj
+                        } else {
+                          return a
+                        }
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx:168]: do not write a as value name, think of some variable name that will define the object

Comment (ArthurO2002) [reply src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx:168]: no else needed

File: src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx [hunk 13]
```diff
@@ -0,0 +1,312 @@
...
+                />
+                <TextField
+                  onChange={(e) => {
+                    const tempObj = {...newSubFilter, name: e.target.value}
+                    const newArr = newSubFilters.map((item: ISubCriteria) => {
+                      if (item.id === tempObj.id) {
+                        return tempObj
+                      } else {
+                        return item
+                      }
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx:262]: no else needed

File: src/pages/events/createEvents/createEventCriteria/CriteriasPaper.tsx [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+    const getRatings = async () => {
+      try {
+        setIsLoading(true)
+        const response: AxiosResponse = await axiosData.get('/rating')
+        await setRatingScoreArr(response.data)
+        setIsLoading(false)
+      } catch (err) {
+        setIsLoading(false)
+      }
+    }
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/createEventCriteria/CriteriasPaper.tsx:66]: Use 1 loading state for both, and 1 try catch

File: src/pages/events/createEvents/createEventCriteria/CriteriasPaper.tsx [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+    getCriterias()
+    getRatings()
+  }, [])
+
+  const onBackButtonClick = () => {
+    dispatchContext({
+      type: createEventReducerTypes.updateActivePageToCriterias,
+      activePage: 'firstPage',
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/createEventCriteria/CriteriasPaper.tsx:74]: Can we have enum values for things like this?

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/pages/events/createEvents/createEventCriteria/BonusPercentage.tsx (hunk 7 above)
Comment (gor017) [reply src/pages/events/createEvents/createEventCriteria/BonusPercentage.tsx]: done

File: src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx (hunk 8 above)
Comment (gor017) [reply src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx]: done

File: src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx (hunk 9 above)
Comment (gor017) [reply src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx]: done

File: src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx (hunk 11 above)
Comment (gor017) [reply src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx]: done

File: src/pages/events/createEvents/createEventCriteria/CriteriasPaper.tsx (hunk 14 above)
Comment (gor017) [reply src/pages/events/createEvents/createEventCriteria/CriteriasPaper.tsx]: done

File: src/pages/events/createEvents/createEventCriteria/CriteriasPaper.tsx (hunk 15 above)
Comment (gor017) [reply src/pages/events/createEvents/createEventCriteria/CriteriasPaper.tsx]: done

**Pull request merged**